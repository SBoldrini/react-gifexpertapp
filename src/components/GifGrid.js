import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs( category );



  return (
    <>
      <h3 id="grid-title" className="animate__animated animate__fadeIn">{ category.toUpperCase() }</h3>
      <hr />

      { loading && <p className="animate__animated animate__flash">Loading...</p> }

      <div className="card-grid">
        {
          images.map(img => (
            <GifGridItem
              key = { img.id }
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}
