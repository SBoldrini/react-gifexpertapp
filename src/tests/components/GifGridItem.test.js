import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Prueba del componente <GifGridItem />', () => {
  
  const title = 'One Punch Man Bubblegum GIF';
  const url = 'https://punchman.gif';
  const wrapper = shallow(<GifGridItem url={ url } title={ title } />);

  test('Debe renderizar correctamente el componente <GifGridItem />', () => {

    expect(wrapper).toMatchSnapshot();

  });

  test('Debe de tener un pÃ¡rrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });
  
  
  test('Debe de tener la imagen ingual al url y alt de los props', () => {
    const img = wrapper.find('img');
    console.log(img.props().alt);

    expect(img.props().alt).toBe(title);
    expect(img.prop('src')).toBe(url);
  });

  test('Debe tener la clase animate__fadeIn', () => {
    const div = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  })
  
  
  
  
});
