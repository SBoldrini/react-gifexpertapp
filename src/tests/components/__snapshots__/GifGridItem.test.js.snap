// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente <GifGridItem /> Debe renderizar correctamente el componente <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="One Punch Man Bubblegum GIF"
    src="https://punchman.gif"
  />
  <p>
    One Punch Man Bubblegum GIF
  </p>
</div>
`;

exports[`Prueba en el componente <GifGridItem /> Debe renderizar correctamente el componente <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="One Punch Man Bubblegum GIF"
    src="https://punchman.gif"
  />
  <p>
    One Punch Man Bubblegum GIF
  </p>
</div>
`;
